src/clients/repos.js

```diff
@@ -0,0 +1,122 @@
+import trim from 'lodash/trim';
+import gitHub from '../services/gitHub';
+import generatePreview from '../util/generatePreview';
+import performWithRetries from '../util/performWithRetries';
+
+const COMMIT_MESSAGE = 'Created using Popcode: https://popcode.org';
+const MASTER_BRANCH = 'master';
+
+// Sometimes there are race conditions with creating multiple files in separate
+// API requests
+function repoPerformWithRetries(perform) {
```

Comment:
just went with `performWithRetryNotFoundOrNetworkErrors`, combined with the feedback below about also retrying on Network.